ash/src/extensions/ext/extended_dynamic_state.rs

```diff
@@ -6,23 +6,17 @@ use std::ptr;
 
 #[derive(Clone)]
 pub struct ExtendedDynamicState {
-    handle: vk::Instance,
-    extended_dynamic_state_fn: vk::ExtExtendedDynamicStateFn,
+    handle: vk::Device,
```

Comment:
Should we be storing handles that aren't used by any of the functions? This seems to occur for a number of extensions.

Reply:
Indeed, I don't think we should store it (beyond using it in the constructor to get device-specific functions) if none of the commands need it. IMO it's fine to just remove `fn instance()` in this case without replacing it with `fn device()` unnecessarily - no need to be "consistent" for that across extension helpers.

After all, it seems `fn instance()` and `fn device()` were added for convenience some time ago, but I would be totally fine to just remove these getters. I doubt they add much, the instance/device is kept for internal use and if it is needed outside of that the caller probably wants to have have the full `ash::Device` (or another extension - which undoubtedly also carries the instance/device where necessary) instead.

Reply:
That sounds good to me as well, though in the interest of controlling scope creep, it'd be fine with me to leave for future work the removal of the getter from cases which do actually need to store a handle.

Reply:
Then `fn fp()` can go to ?

Reply:
Unlike `instance` and `device` the caller doesn't have access to the function pointers which are loaded in the constructor, I'd keep `fn fp()` to access those.

That said though these function pointers are only used circumvent the "overhead" induced by the headers, and eventual functions that did not have their helper counterpart implemented (which should be unlikely / not present at all).

I'd keep them for now and discuss the removal of `device()`/`instance()`/`fp()` in a future issue/PR. The only thing I'd remove in this PR is the `device` and `instance` members (and their matching getters, of course) if they are not used by any of the helper functions.

Reply:
I think it does not make sense to expose `device()`/`instance()` based on wether they are used internally or not.
They should either be always exposed for all extensions or never but not an in between.

Reply:
This is not about exposing those getter functions but about unnecessarily storing these fields in the first place, increasing struct size (concerns have been raised over this in the past, specifically for embedded applications).

Reply:
I'll remove them then.